src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -284,11 +284,14 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,
                                       parameters=parameters, mode=mode, on_error_deployment=on_error_deployment)
 
     smc = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, aux_subscriptions=aux_subscriptions)
+
+    validation_result = smc.deployments.validate(resource_group_name=resource_group_name, deployment_name=deployment_name, properties=properties)
+
+    if validation_result and validation_result.error:
+        raise CLIError(validation_result.error)
     if validate_only:
-        response = sdk_no_wait(no_wait, smc.deployments.validate, resource_group_name, deployment_name, properties)
-        if response and response.error:
-            raise CLIError(response.error)
-        return response
+        return validation_result
```

Comment:
you don't want to support no wait for validate command, right? may I know the reason?

Reply:
this has not changed.. it's the same as we do for `az deployment group validate`. I think the reason might be that the /validate API is not a long running operation, so no-wait is not needed. Let me know if my understanding is not correct.

Reply:
Yes. Thanks for explanation. I have checked the sdk and it is not long running operation.